### ISSUE TITLE
Sites load indefinitely until window is maximized/unmaximized

### ISSUE THREAD

**Issue by HYBRID-BEING** [MrAlex94/Waterfox#1505]
I have a rather peculiar bug, which i'm not even sure how to troubleshoot - all sites i try are loading infinitely. However, if i maximize/unmaximize Waterfox window, sites start loading, but only for a few seconds. After that. sites are again stuck on infinite loading. That started happening overnight (a few days ago) - one evening everything worked fine, next morning - nothing loads normally.
It's not related to update, as i tried updating to 2020.03.1 after it started (didn't help, and update itself wasn't downloading normally). 
I've tried looking at network monitor:
After i refresh page - it shows a single gray GET request for the main document.
After i max/unmax window a few times and page loads - Timings tab for this request shows that request was "Blocked" for a while.

**To Reproduce**
Restart Waterfox
Wait a bit and see that page you start with never finishes loading
Unmaximize Waterfox window

**Expected behavior**
Sites load normally.

**Actual behavior**
Sites don't load unless Waterfox window is maximized/unmaximized.

**My configuration**
Device: PC
OS: Windows 7 64 bit
Version: Classic 2020.03.1

**Comment by Squall-Leonhart** [comment]
No repro, waterfox classic loads everything just fine.

**Comment by HYBRID-BEING** [comment]
Yeah, somehow i doubt i'll find anyone who would be able to reproduce this, so i really could use some advice on how to troubleshoot it.
I also forgot to mention that i've tried safe mode - no dice.

**Comment by Squall-Leonhart** [comment]
reset tcp and winsock, remove any security software that hooks into the browser.

**Comment by HYBRID-BEING** [comment]
I've tried some more things before i had to uninstall my antivirus, and found out a bit more.
I've saved a session with all my tabs (and i hoard them - about 2k currently), opened a new window and closed the last one. Turns out having less tabs, made bug go away temporarily, and pages started to load normally.
However, after reloading session it took different amount of tabs to remove each time to make bug disappear. I've repeated this a few times, but couldn't figure out any regularity. Then i tried restarting Waterfox with an empty window, and reload session after that, and what do you know - the bug seems to be gone for good now.

The bug is peculiar and it went as inexplicably as it appeared, but i think i can't count it as fixed. In case something was really blocking connection browser should at the very least time-out instead of infinite loading. Now that i think about it, i might have had something like this when i used Firefox, so it could be an underlying bug.

**Comment by Squall-Leonhart** [comment]
Can you check the GDI object count for the process?

I'm assuming you're using single process at this point because multi tab processes wouldn't hit a GDI count limit and experience this.

**Comment by HYBRID-BEING** [comment]
Nah, i'm using multiple processes. GDIView shows total of 175 GD objects, with one process having 106 (and that's after loading a few tabs).
I'll try to keep this in mind if bug will resurface, though.

**Comment by Squall-Leonhart** [comment]
its not that then hmm. weird.

**Comment by HYBRID-BEING** [comment]
Welp, bug resurfaced on my 64-bit Windows 10 laptop (2020.04). I've checked GDI count with Process Explorer - about 80 for all waterfox.exe processes. Another thing that i've found interesting is a huge page fault count - it was jumping between 5000 and 50000, usually around 20000-30000, but never getting below 100 (as it does now, after bug again went away).
I've got as much tabs on laptop as on PC, although they are split in 3 windows. I've left it overnight, and, after i minimized two of the windows, the third one started loading (and so did the other two after that). Seems to be working after restart as well. Also, not sure if it's related or not, but bug reappeared after i installed latest Windows 10 updates.
